JetNews/app/src/main/java/com/example/jetnews/ui/article/ArticleScreen.kt

```diff
@@ -64,30 +65,33 @@ import com.example.jetnews.data.posts.impl.BlockingFakePostsRepository
 import com.example.jetnews.data.posts.impl.post3
 import com.example.jetnews.model.Post
 import com.example.jetnews.ui.components.InsetAwareTopAppBar
+import com.example.jetnews.ui.components.JetnewsNavRail
 import com.example.jetnews.ui.home.BookmarkButton
 import com.example.jetnews.ui.theme.JetnewsTheme
 import com.example.jetnews.utils.isScrolled
-import com.example.jetnews.utils.supportWideScreen
 import com.google.accompanist.insets.navigationBarsPadding
 import kotlinx.coroutines.runBlocking
 
 /**
  * Displays the Article screen.
  *
  * @param articleViewModel ViewModel that handles the business logic of this screen
+ * @param showNavRail (state) whether the Drawer or NavigationRail needs to be shown
  * @param onBack (event) request back navigation
  */
 @Composable
 fun ArticleScreen(
     articleViewModel: ArticleViewModel,
+    showNavRail: Boolean,
     onBack: () -> Unit
 ) {
     // UiState of the ArticleScreen
     val uiState by articleViewModel.uiState.collectAsState()
 
     if (uiState.post != null) {
-        ArticleScreen(
+        ArticleAdaptiveScreen(
```

Comment:
I agree that it probably makes sense to keep this as just `ArticleScreen`, `Adaptive` should be the rule, not the exception! The `Route`/`Screen` naming difference should help distinguish between these overloads though.

Reply:
Makes sense! Thank you